src/charm.py

```diff
@@ -195,76 +141,12 @@ def _external_url(self) -> str:
 
     @property
     def _internal_url(self) -> str:
-        scheme = "https" if self.tls_available else "http"
-        return f"{scheme}://{self.hostname}"
-
-    @property
-    def tls_available(self) -> bool:
-        """Return True if tls is enabled and the necessary certs are found."""
-        return (
-            self.cert_handler.enabled
-            and (self.cert_handler.server_cert is not None)
-            and (self.cert_handler.private_key is not None)
-            and (self.cert_handler.ca_cert is not None)
-        )
+        """Returns workload's FQDN."""
+        scheme = "http"
+        if hasattr(self, "coordinator") and self.coordinator.nginx.are_certificates_on_disk:
```

Comment:
I added the `hasattr(...)` to places where `@trace_charm` calls them since they might be called before the `self.coordinator` gets instantiated in `__init__`. I might be wrong. WDYT?

Reply:
`trace_charm` always calls `__init__` first, so this should not be necessary.

Reply:
(if you look at the previous code, we'd have had the same problem with `self.tls_available`)